src/main/java/net/sf/jabref/logic/importer/fetcher/AdsFetcher.java

```diff
@@ -0,0 +1,111 @@
+package net.sf.jabref.logic.importer.fetcher;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Optional;
+
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+
+import net.sf.jabref.logic.help.HelpFile;
+import net.sf.jabref.logic.importer.FetcherException;
+import net.sf.jabref.logic.importer.IdBasedFetcher;
+import net.sf.jabref.logic.importer.ImportFormatPreferences;
+import net.sf.jabref.logic.importer.fileformat.BibtexParser;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.model.entry.BibEntry;
+import net.sf.jabref.model.entry.FieldName;
+
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.http.exceptions.UnirestException;
+import org.apache.http.client.utils.URIBuilder;
+
+/**
+ * This class handles accessing and obtaining BibTeX entry
+ * from ADS(The NASA Astrophysics Data System).
+ * Fetching using DOI(Document Object Identifier) is only supported.
+ */
+public class AdsFetcher implements IdBasedFetcher {
+
+    private static final String URL_PATTERN = "http://adsabs.harvard.edu/doi/";
+
+    private ImportFormatPreferences preferences;
+
+    public AdsFetcher(ImportFormatPreferences preferences) {
+        this.preferences = preferences;
+    }
+
+    @Override
+    public String getName() {
+        return "ADS from ADS-DOI";
+    }
+
+    @Override
+    public HelpFile getHelpPage() {
+        return HelpFile.FETCHER_ADS;
+    }
+
+    @Override
+    public Optional<BibEntry> performSearchById(String identifier) throws FetcherException {
+        Optional<BibEntry> result = Optional.empty();
+
+        String key = identifier.replaceAll("^(doi:|DOI:)", "");
+
+        try {
+            URIBuilder uriBuilder = new URIBuilder(URL_PATTERN + key);
+            uriBuilder.addParameter("data_type", "BIBTEX");
```

Comment:
If you pass "BIBTEXPLUS" as the data_type, then the abstract is also included and the second request below is unnecessary.